### PR TITLE
[feature] add EnCodec model embeddings to FAD calculation

### PR DESCRIPTION
**Pull request by ivanlmh**
Hi Hao Hao, I've added Meta's EnCodec model (the 2 main versions they offer pre-trained) for FAD calculation

- [feature] add EnCodec to available models
- [test] add 24khz and 48khz models' test to test_all notebook
- [update] requirements

Let me know if there is something you think should be fixed or whatever :)

### CODE REVIEW

File: frechet_audio_distance/fad.py [hunk 1]
```diff
@@ -32,7 +34,8 @@ def load_audio_task(fname, sample_rate, dtype="float32"):
         wav_data = wav_data / float(2**31)
 
     # Convert to mono
-    if len(wav_data.shape) > 1:
+    assert channels in [1, 2], "channels must be 1 or 2"
```
Comment (gudgud96) [inline frechet_audio_distance/fad.py:37]: `channels` here is unspecified, I guess you miss a line before this to calculate`channels` from `wav_data`

File: frechet_audio_distance/fad.py [hunk 2]
```diff
@@ -309,8 +360,8 @@ def update(*a):
         for fname in os.listdir(dir):
             res = pool.apply_async(
                 load_audio_task,
-                args=(os.path.join(dir, fname), self.sample_rate, dtype),
-                callback=update
+                args=(os.path.join(dir, fname), self.sample_rate, self.channels, dtype),
```
Comment (gudgud96) [inline frechet_audio_distance/fad.py:363]: I suspect `self.channels` is not defined here, but please correct me if I am wrong.

**Comment by gudgud96** [comment]
@ivanlmh Huge thanks for you contribution. Encodec is a very useful one.
Minor comments related to undefined variables.

File: frechet_audio_distance/fad.py (hunk 2 above)
Comment (ivanlmh) [reply frechet_audio_distance/fad.py:364]: yep, it looks like I erased the definition when commiting

File: frechet_audio_distance/fad.py (hunk 1 above)
Comment (ivanlmh) [reply frechet_audio_distance/fad.py:37]: Yes, it's fixed now. It's an input to the function, defaulting to 1 unless specified (as should be done for encodec 48khz)

**Comment by ivanlmh** [comment]
I think that's fixed... idk why the definitions disappeared, but anyway :)

**Comment by gudgud96** [comment]
@ivanlmh Merged, huge thanks for your contribution once again!

**Pull request merged**